### ISSUE TITLE
Pre-compile macros in the editor version of the engine compilation

### ISSUE THREAD

**Issue by jareguo** [cocos/cocos-engine#15958]
The current engine under the editor (bundled/index.js) should be editor-specific, right? That should have all the macros that can be cured pre-compiled into it, not dynamically judged at runtime. Pre-compiling would be a bigger boost to load performance and runtime performance of the whole scene.
![image](https://github.com/cocos/cocos-engine/assets/1503156/134a0824-5e89-46f2-8df4-ad269b2f5cad)
<img width="550" alt="image" src="https://github.com/cocos/cocos-engine/assets/1503156/8a167c2b-d99a-47d6-bea5-da50a9cca359">
<img width="317" alt="image" src="https://github.com/cocos/cocos-engine/assets/1503156/1d859e89-3840-4d19-884d-dba6bcf20fc3">

**Comment by PPpro** [comment]
> The current engine under the editor (bundled/index.js) should be editor-specific, right? 

not exactly, the current editor engine should also support all editor process, including assetdb, scene, main process and so on, also we need to support game view

**Comment by jareguo** [comment]
I think most of the macros can be cured, and EDITOR is actually just as true in the current game view.

**Comment by dumganhar** [comment]
No plans for this right now.
